class.jetpack-cli.php

```diff
@@ -1034,6 +1034,186 @@ public function authorize_user( $args, $named_args ) {
 		}
 	}
 
+	/**
+	 * Allows management of publicize connections.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <list|disconnect>
+	 * : The action to perform.
+	 * ---
+	 * options:
+	 *  - list
+	 *  - disconnect
+	 * ---
+	 *
+	 * [<identifier>]
+	 * : The connection ID or service to perform an action on.
+	 *
+	 * [--format=<format>]
+	 * : Allows overriding the output of the command when listing connections.
+	 * ---
+	 * default: table
+	 * options:
+	 *  - table
+	 *  - json
+	 *  - csv
+	 *  - yaml
+	 *  - ids
+	 *  - count
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 * wp jetpack publicize list
+	 * wp jetpack publicize list twitter
+	 * wp --user=1 jetpack publicize list
+	 * wp --user=1 jetpack publicize list twitter
+	 * wp jetpack publicize list 123456
+	 * wp jetpack publicize disconnect 123456
+	 * wp jetpack publicize disconnect all
+	 * wp jetpack publicize disconnect twitter
+	 */
+	public function publicize( $args, $named_args ) {
+		if ( ! Jetpack::is_active() ) {
+			WP_CLI::error( __( 'Jetpack is not currently connected to WordPress.com', 'jetpack' ) );
+		}
+
+		$action        = $args[0];
+		$publicize     = new Publicize();
+		$identifier    = ! empty( $args[1] ) ? $args[1] : false;
+		$services      = array_keys( $publicize->get_services() );
+		$id_is_service = in_array( $identifier, $services, true );
+
+		switch ( $action ) {
+			case 'list':
+				$connections_to_return = array();
```

Comment:
All this nesting makes it a bit hard to see what the flow is. I think it would be worth factoring out the `list` and `disconnect` code as separate methods.